src/server/controller.js

```diff
@@ -1,15 +1,64 @@
 const productsMOCK = require('../assets/mocks/products.json');
 
-// const cacheMap = new Map();
+const subscriptions = new Set();
 
 module.exports = {
     getProducts,
     getProductById,
+    getHomepage,
+    addSubscription,
+    deleteSubscription,
     notFound,
 }
 
+
+function addSubscription(req, res) {
+    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Comment:
in current implementation, regExp object will be instantiated  from scratch per every request.

move this pattern out of this function
```
const EMAIL_PATTERN = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Reply:
done